### ISSUE TITLE
The use of Delay()

### ISSUE THREAD

**Issue by whizyrel** [bportaluri/WiFiEsp#217]
You would agree with me that the `delay()` function messes stuff up. Is it avoidable? Are there other alternatives?
Occurrences below
https://github.com/bportaluri/WiFiEsp/blob/9997f650db2322ccc2ef652836196dff253c3f9c/src/WiFiEspClient.cpp#L128
https://github.com/bportaluri/WiFiEsp/blob/9997f650db2322ccc2ef652836196dff253c3f9c/src/WiFiEspClient.cpp#L284

**Comment by JAndrassy** [comment]
yes. my WiFiEspAT library doesn't have them

**Comment by whizyrel** [comment]
Look at the permalink, it seem it does. I added permalinks to where they are in the WiFiEspClient.cpp file

**Comment by JAndrassy** [comment]
> Look at the permalink, it seem it does. I added permalinks to where they are in the WiFiEspClient.cpp file

what permalink? I see only links into the WiFiEsp library source

**Comment by whizyrel** [comment]
Exactly, if you follow the link, you would find the `delay(4000)` on there.

**Comment by JAndrassy** [comment]
> Exactly, if you follow the link, you would find the `delay(4000)` on there.

I was talking about my newer WiFiEspAT library, not about WiFiEsp

**Comment by whizyrel** [comment]
Oh, Can you point me in the direction of where that is

**Comment by whizyrel** [comment]
I think I found it https://github.com/jandrassy/WiFiEspAT

**Comment by whizyrel** [comment]
Quick question, do you think the WiFiEspAT can work for this?
<--- 'AR wersion:1.2.0.0(Jul  1 2016 20:04:45)'
14:31:17.347 -> AR wersion:1.2.0.0(Jul  1 2016 20:04:45)
14:31:17.347 -> SDK version:1.5/4.1(39cb9a32)
14:31:17.347 -> v1.0.0
14:31:17.347 -> Mar 11 2018 18:27:31
14:31:17.347 -> O'

**Comment by whizyrel** [comment]
Help!!!

**Comment by JAndrassy** [comment]
https://github.com/jandrassy/WiFiEspAT#why-a-new-wifiesp-library
